src/link/MachO/Zld.zig

```diff
@@ -1833,9 +1833,13 @@ fn resolveSymbols(self: *Zld) !void {
         var found = false;
         for (self.dylibs.items) |dylib| {
             const proxy = dylib.symbols.get(undef.name) orelse continue;
-            try self.imports.putNoClobber(self.allocator, proxy.name, proxy);
-            undef.alias = proxy;
-            found = true;
+            log.warn("{s}", .{proxy.name});
+            if(self.imports.contains(proxy.name))
```

Comment:
Proxies should never collide meaning `self.imports.putNoClobber(...)` should never trap. If that is the case, the linking will most likely result in missing links between `Undef`s and `Proxy`s.

Reply:
Either way, this logic is completely rewritten as part of #9171 so you might wanna wait for this to land before tweaking the symbol resolution logic.

Reply:
See the comment below: I'm surprised you need to touch the symbol resolution logic at all.

Reply:
Yeah - it's worth pointing out that this was just a short-term hack to just get an example working. I'm getting some kind of conflict with the library I've been testing with - I just don't know why & this was a very hacky way to just work around that.

Reply:
So the reason I added that log warnings was to see why that was happening - and every proxy.name appears twice (back-to-back) for some reason. This is what i get when linking with the vulkan dynamic libraries:

```
warning(zld): _vkQueuePresentKHR
warning(zld): _vkQueuePresentKHR
warning(zld): _vkGetSwapchainImagesKHR
warning(zld): _vkGetSwapchainImagesKHR
warning(zld): _vkGetPhysicalDeviceSurfaceSupportKHR
warning(zld): _vkGetPhysicalDeviceSurfaceSupportKHR
warning(zld): _vkGetPhysicalDeviceSurfaceFormatsKHR
warning(zld): _vkGetPhysicalDeviceSurfaceFormatsKHR
warning(zld): _vkGetPhysicalDeviceSurfaceCapabilitiesKHR
warning(zld): _vkGetPhysicalDeviceSurfaceCapabilitiesKHR
warning(zld): _vkGetPhysicalDeviceQueueFamilyProperties
warning(zld): _vkGetPhysicalDeviceQueueFamilyProperties
warning(zld): _vkGetPhysicalDeviceMemoryProperties
warning(zld): _vkGetPhysicalDeviceMemoryProperties
warning(zld): _vkGetDeviceQueue
warning(zld): _vkGetDeviceQueue
warning(zld): _vkFreeMemory
warning(zld): _vkFreeMemory
warning(zld): _vkEnumeratePhysicalDevices
warning(zld): _vkEnumeratePhysicalDevices
warning(zld): _vkEndCommandBuffer
warning(zld): _vkEndCommandBuffer
warning(zld): _vkDestroySwapchainKHR
warning(zld): _vkDestroySwapchainKHR
warning(zld): _vkDestroyInstance
warning(zld): _vkDestroyInstance
warning(zld): _vkDestroyDevice
warning(zld): _vkDestroyDevice
warning(zld): _vkDestroyDescriptorPool
warning(zld): _vkDestroyDescriptorPool
warning(zld): _vkDestroyCommandPool
warning(zld): _vkDestroyCommandPool
warning(zld): _vkCreateSwapchainKHR
warning(zld): _vkCreateSwapchainKHR
warning(zld): _vkCreateShaderModule
warning(zld): _vkCreateShaderModule
warning(zld): _vkCreateSemaphore
warning(zld): _vkCreateSemaphore
warning(zld): _vkCreatePipelineLayout
warning(zld): _vkCreatePipelineLayout
warning(zld): _vkCreatePipelineCache
warning(zld): _vkCreatePipelineCache
warning(zld): _vkCreateInstance
warning(zld): _vkCreateInstance
warning(zld): _vkCreateImageView
warning(zld): _vkCreateImageView
warning(zld): _vkCreateDevice
warning(zld): _vkCreateDevice
warning(zld): _vkCreateDescriptorSetLayout
warning(zld): _vkCreateDescriptorSetLayout
warning(zld): _vkCreateDescriptorPool
warning(zld): _vkCreateDescriptorPool
warning(zld): _vkCreateComputePipelines
warning(zld): _vkCreateComputePipelines
warning(zld): _vkCreateCommandPool
warning(zld): _vkCreateCommandPool
warning(zld): _vkCmdBindPipeline
warning(zld): _vkCmdBindPipeline
warning(zld): _vkCmdBindDescriptorSets
warning(zld): _vkCmdBindDescriptorSets
warning(zld): _vkBeginCommandBuffer
warning(zld): _vkBeginCommandBuffer
warning(zld): _vkAllocateMemory
warning(zld): _vkAllocateMemory
warning(zld): _vkAllocateDescriptorSets
warning(zld): _vkAllocateDescriptorSets
warning(zld): _vkAllocateCommandBuffers
warning(zld): _vkAllocateCommandBuffers
warning(zld): _vkAcquireNextImageKHR
warning(zld): _vkAcquireNextImageKHR
warning(zld): _vkResetCommandPool
warning(zld): _vkResetCommandPool
warning(zld): _vkUpdateDescriptorSets
warning(zld): _vkUpdateDescriptorSets
```

Reply:
Oh haha - I've also just realised that the if logic I intended was inverted - whoops.

Reply:
Yep - that explains why the built binary never actually ran properly (I only had the compile "successfully" complete) - no that I've actually got the if statement doing what I intended I do now have a functioning executable! :D 

However- now I need to actually solve the problem that is causing a need for this (because it should not be needed).

Reply:
OK - I'm done for the day, so will investigate this issue another time. If you have any ideas of what might be causing these duplicate symbols please do let me know. Basically - if I don't have this if statement - I get a panic in putNoClobber.

Reply:
I think you might wanna rebase onto `zld-tapi` locally, remove any tweaks to symbol resolutions, and see if it works fine then. Kinda hoping that changes in that branch made the symbol resolution logic more robust.

Reply:
Yes! That seems to resolve the issue. Just running the built executable on a Mac now to check that it works. 

If this is the thing that solves this then - what would you like me to do w.r.t this branch/PR relative to the `zld-tapi` branch?

Reply:
The built executable ran successfully! :) I've pushed the removal of the ZLD changes to the branch here - but let me know what would be good for me to do regarding this.

Also if you have any further feedback on the current state of the PR that would be really helpful - I believe I've resolved some of it, but as this is my first real chunk of zig code that I've written (outside of a build.zig file for me C++ project), I'm aware it might be a bit idiosyncratic. 

If it does reach the point where from a code-POV it's an OK set of changes to merge, would you also like me to squash the commits?

Reply:
I've squashed the commits (because it felt like thee right thing to do) - the only thing I'm still unsure about is if this ready to merge as-is (because the changes were only actually functional for me when merged with the zld-tapi branch). What is the eta for the zld-tapi branch to be merged into master?

Reply:
`zld-tapi` landing in master should happen really soon.